### PR TITLE
Update Desktop NUSPEC to use ARM64EC

### PR DESCRIPTION
**Pull request by JunielKatarn**
## Description
Add ARM64EC platform to Desktop NuGet.

### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
Currently the Desktop NUSPEC `OfficeReact.Win32.nuspec` expects source and target paths for ARM64.

### What
Change the NUSPEC entries from `ARM64` to `ARM64EC`.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/13335)

### CODE REVIEW

**Review by vmoroz — APPROVED** [review]

**Review by dannyvv — APPROVED** [review]

File: .ado/publish.yml [hunk 1]
```diff
@@ -492,13 +492,13 @@ extends:
                   configuration: Release
                 - platform: x86
                   configuration: Release
-                - platform: ARM64
+                - platform: ARM64EC
```
Comment (jeffstall) [inline .ado/publish.yml:495]: I think we need to have __both__ ARM64 and ARM64EC builds. We can't replace one with the other. There's usage cases for both. Can you please update?

**Review by jeffstall — CHANGES_REQUESTED** [review]
I think we need to have both ARM64 and ARM64EC builds. We can't replace one with the other. There's usage cases for both. Can you please update?

File: .ado/publish.yml (hunk 1 above)
Comment (JunielKatarn) [reply .ado/publish.yml:495]: > I think we need to have **both** ARM64 and ARM64EC builds. We can't replace one with the other. There's usage cases for both. Can you please update?

This change is exclusive for the Desktop DLL (`react-native-win32.dll`). See `artifactName: Desktop`
We do not need a pure `ARM64` build for this variant.

This should not affect the UWP build of RNW.

File: vnext/Scripts/OfficeReact.Win32.nuspec [hunk 2]
```diff
@@ -15,25 +15,25 @@
...
-    <file src="$nugetroot$\x64\Debug\React.Windows.Desktop\Microsoft.ReactNative.winmd" target="lib\debug\x64"/>
-    <file src="$nugetroot$\ARM64\Debug\React.Windows.Desktop\Microsoft.ReactNative.winmd" target="lib\debug\ARM64"/>
-    <file src="$nugetroot$\x86\Release\React.Windows.Desktop\Microsoft.ReactNative.winmd" target="lib\ship\x86"/>
-    <file src="$nugetroot$\x64\Release\React.Windows.Desktop\Microsoft.ReactNative.winmd" target="lib\ship\x64"/>
-    <file src="$nugetroot$\ARM64\Release\React.Windows.Desktop\Microsoft.ReactNative.winmd" target="lib\ship\ARM64"/>
+    <file src="$nugetroot$\x86\Debug\React.Windows.Desktop.DLL\react-native-win32.*"          target="lib\debug\x86"      exclude="**\*.iobj;**\*.ipdb;**\*.exp;**\*.ilk" />
+    <file src="$nugetroot$\x64\Debug\React.Windows.Desktop.DLL\react-native-win32.*"          target="lib\debug\x64"      exclude="**\*.iobj;**\*.ipdb;**\*.exp;**\*.ilk" />
+    <file src="$nugetroot$\ARM64EC\Debug\React.Windows.Desktop.DLL\react-native-win32.*"      target="lib\debug\ARM64EC"  exclude="**\*.iobj;**\*.ipdb;**\*.exp;**\*.ilk" />
```
Comment (jeffstall) [inline vnext/Scripts/OfficeReact.Win32.nuspec:20]: I think that removing ARM64 and replacing with ARM64EC is a problem. I think we need both ARM64 (native) and ARM64EC as two different, parallel options. Can you check again?

**Review by jeffstall — CHANGES_REQUESTED** [review]

File: vnext/Scripts/OfficeReact.Win32.nuspec (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Scripts/OfficeReact.Win32.nuspec:20]: I have checked again.
There is no reported usage of `react-native-win32.dll` by any products other than Office, which is only built as ARM64EC for the corresponding hardware architecture.

Therefore, the ARM64 variant is not needed.

**Review by dannyvv — APPROVED** [review]

**Review by vmoroz — APPROVED** [review]

**Pull request merged**